### PR TITLE
feat: dashboard API of task node

### PR DESCRIPTION
**Pull request by STRRL**
Signed-off-by: STRRL <str_ruiling@outlook.com>

### What problem does this PR solve?

Issue Number: It's a part of https://github.com/chaos-mesh/chaos-mesh/issues/2051 .

### What is changed and how it works?

What's Changed:

- return `conditionalBranches` in the Dashboard API.

### Related changes

* PR to update `chaos-mesh/website`/`chaos-mesh/website-zh`: No.
* Need to update Chaos Dashboard component, related issue: Yes, will be tracked in the issue #2051.
* Need to cheery-pick to the release branch. Yes.

### Checklist

Tests

- [x] Unit test
- [ ] E2E test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Breaking backward compatibility

### Release note 

```release-note
Please add a release note.
If you don't think this PR needs a release note then fill it with None.
```

### CODE REVIEW

**Comment by STRRL** [comment]
CI fixed! 

PTAL @g1eny0ung

File: pkg/core/workflow.go [hunk 1]
```diff
@@ -82,20 +82,23 @@ const (
...
+	Template            string              `json:"template"`
+	UID                 string              `json:"uid"`
 }
 
 type NodeNameWithTemplate struct {
 	Name     string `json:"name,omitempty"`
 	Template string `json:"template,omitempty"`
 }
 
+// FIXME: unnecessary another wrap of []NodeNameWithTemplate, like struct NodeSerial and NodeParallel
```
Comment (g1eny0ung) [inline pkg/core/workflow.go:100]: Could u explain more about this? I'm not sure why it's unnecessary.

**Review by g1eny0ung — APPROVED** [review]

File: pkg/core/workflow.go (hunk 1 above)
Comment (g1eny0ung) [reply pkg/core/workflow.go:100]: I have already approved because this comment won't affect the changes.

Comment (STRRL) [reply pkg/core/workflow.go:100]: related issue https://github.com/chaos-mesh/chaos-mesh/issues/2054, struct `NodeSerial` and `NodeParallel` only contains one field `[]NodeNameWithTemplate`, it looks ugly. :cry:

Comment (g1eny0ung) [reply pkg/core/workflow.go:100]: Got it.

**Review by YangKeao — APPROVED** [review]
LGTM

**Pull request merged**